test/integration/scheduler/framework_test.go

```diff
@@ -126,11 +126,11 @@ var _ = framework.PostbindPlugin(&PostbindPlugin{})
 var _ = framework.UnreservePlugin(&UnreservePlugin{})
 var _ = framework.PermitPlugin(&PermitPlugin{})
 
-var scPlugin = &ScorePlugin{}
-
-// NewScorePlugin is the factory for score plugin.
-func NewScorePlugin(_ *runtime.Unknown, _ framework.FrameworkHandle) (framework.Plugin, error) {
-	return scPlugin, nil
+// newScorePlugin returns a factory for score plugin with specified ScorePlugin.
+func newScorePlugin(scorePlugin *ScorePlugin) framework.PluginFactory {
```

Comment:
You can simply compose a generic `newPlugin`:

```go
func newPlugin(plugin framework.Plugin) framework.PluginFactory {
	return func(_ *runtime.Unknown, fh framework.FrameworkHandle) (framework.Plugin, error) {
		return plugin, nil
	}
}
```

And remove all `newXYZPlugin()` b/c all callers can just pass in specific plugin struct as a plugin interface obj.

Reply:
Addressed. If we need a more specific newXXXPlugin, we could add it later.